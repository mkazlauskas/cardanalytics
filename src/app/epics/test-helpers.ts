export const assertDeepEqual = (actual: unknown, expected: unknown) => {
  expect(actual).toEqual(expected);
};
