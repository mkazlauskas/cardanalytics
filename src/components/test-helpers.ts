import { CardanoStats } from '../app/types';

export const cardanoData: CardanoStats = {
  blocks: 5995095,
  transactions: 9787836,
  assets: 43751,
  epoch: 278,
  slot: 300948,
  currentFees: {
    min: 0.16,
    max: 0.26,
    average: 0.2,
  }
};