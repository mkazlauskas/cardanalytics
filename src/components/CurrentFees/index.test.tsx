import React from 'react';
import { useSelector } from 'react-redux';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Component from '.';
import { cardanoData } from '../test-helpers';

describe('CardanoFees', () => {
  beforeAll(() => {
    // Use mock store data in react-redux hook
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (useSelector as any).mockImplementation(() => cardanoData);
  });
  it('should match snapshot', () => {
    expect(toJson(shallow(<Component />))).toMatchSnapshot();
  });
});