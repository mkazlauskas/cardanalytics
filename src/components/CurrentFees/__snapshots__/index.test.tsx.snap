// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardanoFees should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="column"
>
  <span>
    Min Fee: 0.00 ADA
  </span>
  <span>
    Max Fee: 0.00 ADA
  </span>
  <span>
    Average Fee: 0.00 ADA
  </span>
</WithStyles(ForwardRef(Grid))>
`;
